### PR TITLE
chore: Add filter utility to improve discoverability

### PR DESCRIPTION
**Pull request by KavithaSiva**
This PR introduces `FilterUtility` to make new functionality being added to filter configuration like for e.g a new service provider discoverable.
Still choosing to retain the direct function usage (`azureContentFilter()`) as well because currently we only have one option and it shortens code.

Proposed state:
```Typescript
client.chatCompletion({
  deploymentConfiguration: { deploymentId: 'id' },
  llmConfig: {
    model_name: 'gpt-35-turbo-16k',
    model_params: { max_tokens: 50, temperature: 0.1 }
  },
  prompt: {
    template: [{ role: 'user', content: 'What is the capital of Germany?' }]
  },
  filterConfig: {
     //Both options are allowed
    input: FilterUtility.azureContentFilter({ Hate: 4, SelfHarm: 2 }), 
    output: azureContentFilter({ Sexual: 0, Violence: 6 })
  }
});
```

### CODE REVIEW

**Comment by tomfrenken** [comment]
I like the idea of improved discoverability, however, I really dislike long call chains (e.g. `MyCoolClass.someFunctionForSomething()`) therefore I would be in favor of adding a dedicated section for our filter function(s) as they grow.

**Comment by tomfrenken** [comment]
On the other hand, depending on how the different filters will be chained together, we might even consider a similar implementation to the cloud sdk's resilience package: https://sap.github.io/cloud-sdk/docs/js/guides/resilience

**Comment by KavithaSiva** [comment]
> On the other hand, depending on how the different filters will be chained together, we might even consider a similar implementation to the cloud sdk's resilience package: https://sap.github.io/cloud-sdk/docs/js/guides/resilience

Could you sketch out an example on how this would look?

File: packages/gen-ai-hub/src/orchestration/orchestration-client.test.ts [hunk 1]
```diff
@@ -78,7 +81,7 @@ describe('GenAiHubClient', () => {
       },
       filterConfig: {
         input: azureContentFilter({ Hate: 4, SelfHarm: 2 }),
-        output: azureContentFilter({ Sexual: 0, Violence: 4 })
+        output: FilterUtility.azureContentFilter({ Sexual: 0, Violence: 4 })
```
Comment (jjtang1985) [inline packages/gen-ai-hub/src/orchestration/orchestration-client.test.ts:84]: I prefer the original API.

**Review by jjtang1985 — CHANGES_REQUESTED** [review]
I prefer the original API, which is short and JS native.

**Pull request closed without merging**